src/test/tools/extents/extents.c

```diff
@@ -72,12 +72,53 @@ main(int argc, char *argv[])
 	if (ret)
 		goto exit_free;
 
-	unsigned e;
-	for (e = 0; e < exts->extents_count; e++)
-		printf("%lu %lu\n",
+	if (argc == 2) {
+		/* print out all extents */
+		for (unsigned e = 0; e < exts->extents_count; e++) {
 			/* extents are in bytes, convert them to sectors */
-			B2SEC(exts->extents[e].offset_physical),
-			B2SEC(exts->extents[e].length));
+			printf("%lu %lu\n",
+				B2SEC(exts->extents[e].offset_physical),
+				B2SEC(exts->extents[e].length));
+		}
+	} else {
+		/* print the physical offset of the given logical one */
+		char *error;
+
+		long unsigned offset = strtoul(argv[2], &error, 10 /* base */);
+		if (*error != '\0') {
+			printf(
+				"error: invalid character(s) in the given logical offset: %s\n",
+				error);
+			ret = -1;
+			goto exit_free;
+		}
+
+		unsigned extent = 0;
+		for (unsigned e = 0; e < exts->extents_count; e++) {
+			if (B2SEC(exts->extents[e].offset_logical) > offset)
+				break;
+			extent = e;
+		}
+
+		if (extent == exts->extents_count - 1) {
+			long unsigned max_log;
+
+			max_log = B2SEC(exts->extents[extent].offset_logical) +
+					B2SEC(exts->extents[extent].length);
+
+			if (offset > max_log) {
+				printf("error: maximum logical offset is %lu\n",
```

Comment:
Please either remove this message (because this can be understood ad filtering returning zero entries) or print it to stderr.

Reply:
Done